torch/distributed/rpc/api.py

```diff
@@ -168,7 +173,7 @@ def _wait_all_workers():
 
 
 @_require_initialized
-def shutdown(graceful=True):
+def shutdown(graceful=True, exit_on_worker_done=None):
```

Comment:
I am OK with adding one arg to `_wait_all_workers`, but have a little hesitation in adding it here:

1. It's like we expose the internal termination algorithm to the user API. 
2. Is it true that `exit_on_worker_done` only work with `graceful = True`?
3. What is the implication here if we add elasticity in the future, i.e., allow nodes to join and leave dynamically? (cc @agolynski)

Reply:
Does this actually mean we should make termination algorithm pluggable?

Reply:
> 1. It's like we expose the internal termination algorithm to the user API.
> 2. Is it true that `exit_on_worker_done` only work with `graceful = True`?

> Does this actually mean we should make termination algorithm pluggable?

For 1, Yeah, I am making users have to make a choice between 2 algorithms.
For 2, you are right.

I agree that this is not a clean API yet.

Good idea. I am indeed adding a new shutdown algorithm/strategy. 
It's better to make it pluggable, and pass a per-strategy options struct, like what we do for different RPC backends.

Let me try to extract an interface out of it.